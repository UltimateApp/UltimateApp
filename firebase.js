import * as firebase from 'firebase';

const firebaseConfig = {
  apiKey: 'AIzaSyAwHIKceH-r5UpcpmgiD5PI6PEM3PMn_j0',
  authDomain: 'discinfirebase.firebaseapp.com',
  databaseURL: 'https://discinfirebase-default-rtdb.firebaseio.com',
  projectId: 'discinfirebase',
  storageBucket: 'discinfirebase.appspot.com',
  messagingSenderId: '706426447682',
  appId: '1:706426447682:web:166bcbb53450542aff8adb',
};

firebase.initializeApp(firebaseConfig);
