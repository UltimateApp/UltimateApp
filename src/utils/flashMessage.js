import FlashMessage, { showMessage } from 'react-native-flash-message';

import theme from '../styles/theme.style';
import I18n from './i18n';

export const showSuccess = (description, ref = null) => {
  const params = {
    message: I18n.t('utils.snackbar.success'),
    description,
    type: 'success',
    floating: 'true',
    titleStyle: {
      fontSize: theme.FONT_SIZE_MEDIUM,
    },
    textStyle: {
      fontSize: theme.FONT_SIZE_SMALL,
    },
  };
  ref ? ref.showMessage(params) : showMessage(params);
};

export const showError = (description, ref = null) => {
  const params = {
    message: I18n.t('utils.snackbar.error'),
    description,
    type: 'danger',
    floating: 'true',
    titleStyle: {
      fontSize: theme.FONT_SIZE_MEDIUM,
    },
    textStyle: {
      fontSize: theme.FONT_SIZE_SMALL,
    },
  };
  ref ? ref.showMessage(params) : showMessage(params);
};

export default FlashMessage;
