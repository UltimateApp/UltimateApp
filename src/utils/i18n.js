import I18n from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en';
import fr from './locales/fr';

I18n.locale = Localization.locale;
I18n.fallbacks = true;
I18n.translations = {
  en,
  fr,
};

export default I18n;
