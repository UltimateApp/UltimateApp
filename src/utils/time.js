export const convertMinsToTime = mins => {
  const hours = Math.floor(mins / 60);
  let minutes = mins % 60;
  if (minutes !== 0) {
    minutes = minutes < 10 ? '0' + minutes : minutes;
    return `${hours} h ${minutes}`;
  }
  return `${hours} h`;
};
