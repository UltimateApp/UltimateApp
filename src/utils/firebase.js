import firebase from 'firebase/app';
import 'firebase/database';

//Initialize Firebase
if (firebase.apps.length === 0) {
  firebase.initializeApp(firebaseConfig);
}

const reference = (uuid) => {
  return firebase.database().ref(`customPlays/${uuid}`);
};

export const upload = (play) => {
  return reference(play.uuid).set(play);
};

export const download = (uuid) => {
  return reference(uuid)
    .once('value')
    .then((snapshot) => snapshot.val());
};
