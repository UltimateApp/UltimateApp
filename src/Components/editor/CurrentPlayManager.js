import React, { useState } from 'react';
import { Alert, Share, Text } from 'react-native';
import { Menu, Divider } from 'react-native-paper';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import * as Linking from 'expo-linking';

import I18n from '../../utils/i18n';
import { upload } from '../../utils/firebase';
import { showSuccess, showError } from '../../utils/flashMessage';
import HeaderButton from '../shared/HeaderButton';

const CurrentPlayManager = (props) => {
  const [visible, setVisible] = useState(false);

  const openMenu = () => setVisible(true);
  const closeMenu = () => setVisible(false);

  const share = async () => {
    try {
      await upload(props.currentPlay);
      const url = Linking.makeUrl('customPlays/' + props.currentPlay.uuid);
      await Share.share({
        title: I18n.t('editor.currentPlayManager.shareTitle', { title: props.currentPlay.title }),
        message: I18n.t('editor.currentPlayManager.shareMessage', { url }),
        url,
      });
    } catch (error) {
      showError(I18n.t('editor.currentPlayManager.shareError'));
    }
  };

  const checkBeforeNewPlay = () => {
    if (props.isPlaySaved) {
      props.new();
    } else {
      Alert.alert(
        I18n.t('editor.saveModificationsTitle'),
        I18n.t('editor.saveModificationsText', { title: props.currentPlay.title }),
        [
          {
            text: I18n.t('shared.cancel'),
            style: 'cancel',
            onPress: () => {},
          },
          {
            text: I18n.t('shared.yes'),
            onPress: () => {
              props.save();
              showSuccess(I18n.t('editor.currentPlayManager.saveSuccess', { title: props.currentPlay.title }));
              props.new();
            },
          },
          {
            text: I18n.t('shared.no'),
            onPress: () => {
              props.new();
            },
          },
        ],
      );
    }
  };

  return (
    <Menu visible={visible} onDismiss={closeMenu} anchor={<HeaderButton icon="dots-vertical" onPress={openMenu} />}>
      <Menu.Item
        onPress={() => {
          props.save();
          showSuccess(I18n.t('editor.currentPlayManager.saveSuccess', { title: props.currentPlay.title }));
          closeMenu();
        }}
        icon="content-save-outline"
        title={I18n.t('editor.currentPlayManager.save')}
      />
      <Menu.Item
        onPress={() => {
          props.rename();
          closeMenu();
        }}
        icon="pencil"
        title={I18n.t('editor.currentPlayManager.rename')}
      />
      <Menu.Item
        onPress={() => {
          checkBeforeNewPlay();
          closeMenu();
        }}
        icon="plus"
        title={I18n.t('editor.currentPlayManager.new')}
      />
      <Divider />
      <Menu.Item
        onPress={() => {
          share();
          closeMenu();
        }}
        icon="share-outline"
        title={I18n.t('editor.currentPlayManager.share')}
      />
    </Menu>
  );
};

export default CurrentPlayManager;
