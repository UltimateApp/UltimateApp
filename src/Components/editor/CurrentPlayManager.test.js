import React from 'react';
import { render, waitFor, fireEvent } from '@testing-library/react-native';
import { Provider as PaperProvider } from 'react-native-paper';
import { Alert, Share } from 'react-native';

import animationSquare from '../../Fixtures/Animation/AnimationSquare';
import * as firebase from '../../utils/firebase';

import CurrentPlayManager from './CurrentPlayManager';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<CurrentPlayManager />', () => {
  afterEach(() => jest.clearAllMocks());

  const save = jest.fn();
  const _new = jest.fn();
  const rename = jest.fn();
  let isPlaySaved = true;
  const currentPlay = { animation: animationSquare, title: 'Square', uuid: '123' };

  it('renders correctly', async () => {
    const { toJSON, getByTestId } = await waitFor(() =>
      render(
        <PaperProvider>
          <CurrentPlayManager
            currentPlay={currentPlay}
            isPlaySaved={isPlaySaved}
            save={save}
            new={_new}
            rename={rename}
          />
        </PaperProvider>,
      ),
    );
    expect(toJSON()).toMatchSnapshot();

    await fireEvent.press(getByTestId('headerButton'));

    expect(toJSON()).toMatchSnapshot();
  });

  it('triggers save action', async () => {
    const { getByTestId, getByText } = await waitFor(() =>
      render(
        <PaperProvider>
          <CurrentPlayManager
            currentPlay={currentPlay}
            isPlaySaved={isPlaySaved}
            save={save}
            new={_new}
            rename={rename}
          />
        </PaperProvider>,
      ),
    );
    await fireEvent.press(getByTestId('headerButton'));
    await fireEvent.press(getByText('Save'));

    expect(save).toBeCalled();
  });

  it('triggers rename action', async () => {
    const { getByTestId, getByText } = await waitFor(() =>
      render(
        <PaperProvider>
          <CurrentPlayManager
            currentPlay={currentPlay}
            isPlaySaved={isPlaySaved}
            save={save}
            new={_new}
            rename={rename}
          />
        </PaperProvider>,
      ),
    );
    await fireEvent.press(getByTestId('headerButton'));
    await fireEvent.press(getByText('Rename'));

    expect(rename).toBeCalled();
  });

  describe('New', () => {
    it('triggers new action when current play is saved', async () => {
      const { getByTestId, getByText } = await waitFor(() =>
        render(
          <PaperProvider>
            <CurrentPlayManager
              currentPlay={currentPlay}
              isPlaySaved={isPlaySaved}
              save={save}
              new={_new}
              rename={rename}
            />
          </PaperProvider>,
        ),
      );
      await fireEvent.press(getByTestId('headerButton'));
      await fireEvent.press(getByText('New'));

      expect(_new).toBeCalled();
    });

    it('asks for confirmation when current play is not saved', async () => {
      isPlaySaved = false;
      jest.spyOn(Alert, 'alert');

      const { getByTestId, getByText, debug } = await waitFor(() =>
        render(
          <PaperProvider>
            <CurrentPlayManager
              currentPlay={currentPlay}
              isPlaySaved={isPlaySaved}
              save={save}
              new={_new}
              rename={rename}
            />
          </PaperProvider>,
        ),
      );
      await fireEvent.press(getByTestId('headerButton'));
      await fireEvent.press(getByText('New'));

      // Opens confirmation Alert
      expect(Alert.alert).toHaveBeenCalled();
    });

    it('does nothing on cancel in the confirmation alert', async () => {
      isPlaySaved = false;
      jest.spyOn(Alert, 'alert');

      const { getByTestId, getByText, debug } = await waitFor(() =>
        render(
          <PaperProvider>
            <CurrentPlayManager
              currentPlay={currentPlay}
              isPlaySaved={isPlaySaved}
              save={save}
              new={_new}
              rename={rename}
            />
          </PaperProvider>,
        ),
      );
      await fireEvent.press(getByTestId('headerButton'));
      await fireEvent.press(getByText('New'));

      // Opens confirmation Alert
      expect(Alert.alert).toHaveBeenCalled();

      // Press Cancel
      expect(Alert.alert.mock.calls[0][2][0].text).toEqual('Cancel');
      Alert.alert.mock.calls[0][2][0].onPress();

      expect(save).not.toBeCalled();
      expect(_new).not.toBeCalled();
    });

    it('calles save then new on pressing Yes in the confirmation alert', async () => {
      isPlaySaved = false;
      jest.spyOn(Alert, 'alert');

      const { getByTestId, getByText, debug } = await waitFor(() =>
        render(
          <PaperProvider>
            <CurrentPlayManager
              currentPlay={currentPlay}
              isPlaySaved={isPlaySaved}
              save={save}
              new={_new}
              rename={rename}
            />
          </PaperProvider>,
        ),
      );
      await fireEvent.press(getByTestId('headerButton'));
      await fireEvent.press(getByText('New'));

      // Opens confirmation Alert
      expect(Alert.alert).toHaveBeenCalled();

      // Press Yes
      expect(Alert.alert.mock.calls[0][2][1].text).toEqual('Yes');
      Alert.alert.mock.calls[0][2][1].onPress();

      expect(save).toBeCalled();
      expect(_new).toBeCalled();
    });

    it('calles save then new on pressing No in the confirmation alert', async () => {
      isPlaySaved = false;
      jest.spyOn(Alert, 'alert');

      const { getByTestId, getByText } = await waitFor(() =>
        render(
          <PaperProvider>
            <CurrentPlayManager
              currentPlay={currentPlay}
              isPlaySaved={isPlaySaved}
              save={save}
              new={_new}
              rename={rename}
            />
          </PaperProvider>,
        ),
      );
      await fireEvent.press(getByTestId('headerButton'));
      await fireEvent.press(getByText('New'));

      // Opens confirmation Alert
      expect(Alert.alert).toHaveBeenCalled();

      // Press No
      expect(Alert.alert.mock.calls[0][2][2].text).toEqual('No');
      Alert.alert.mock.calls[0][2][2].onPress();

      expect(save).not.toBeCalled();
      expect(_new).toBeCalled();
    });
  });

  it('triggers share action', async () => {
    jest.spyOn(firebase, 'upload').mockImplementation(() => {});

    const share = jest.fn();
    Share.share = () => new Promise((resolve, reject) => share());

    const { getByTestId, getByText } = await waitFor(() =>
      render(
        <PaperProvider>
          <CurrentPlayManager
            currentPlay={currentPlay}
            isPlaySaved={isPlaySaved}
            save={save}
            new={_new}
            rename={rename}
          />
        </PaperProvider>,
      ),
    );
    await fireEvent.press(getByTestId('headerButton'));
    await fireEvent.press(getByText('Share'));

    expect(firebase.upload).toHaveBeenCalled();
    expect(share).toHaveBeenCalled();
  });
});
