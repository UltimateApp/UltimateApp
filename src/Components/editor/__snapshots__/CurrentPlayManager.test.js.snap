// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentPlayManager /> renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="headerButton"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "fontSize": 28,
              "marginRight": 20,
            },
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<CurrentPlayManager /> renders correctly 2`] = `
Array [
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="headerButton"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "fontSize": 28,
                "marginRight": 20,
              },
              Object {
                "fontFamily": "material-community",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityLabel="Close menu"
      accessibilityRole="button"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityViewIsModal={true}
      collapsable={false}
      onAccessibilityEscape={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "left": 8,
            "top": 8,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateY": -0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "elevation": 8,
              "opacity": 0,
              "paddingVertical": 8,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 7,
                "width": 0,
              },
              "shadowOpacity": 0.24,
              "shadowRadius": 8,
              "transform": Array [
                Object {
                  "scaleX": 0,
                },
                Object {
                  "scaleY": 0,
                },
              ],
            }
          }
        >
          <View
            accessibilityRole="menuitem"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 280,
                    "minWidth": 112,
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "justifyContent": "center",
                      "maxWidth": 264,
                      "minWidth": 96,
                    },
                    Object {
                      "maxWidth": 192,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Save
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityRole="menuitem"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 280,
                    "minWidth": 112,
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "justifyContent": "center",
                      "maxWidth": 264,
                      "minWidth": 96,
                    },
                    Object {
                      "maxWidth": 192,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Rename
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityRole="menuitem"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 280,
                    "minWidth": 112,
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "justifyContent": "center",
                      "maxWidth": 264,
                      "minWidth": 96,
                    },
                    Object {
                      "maxWidth": 192,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  New
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                undefined,
              ]
            }
          />
          <View
            accessibilityRole="menuitem"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                false,
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 280,
                    "minWidth": 112,
                    "paddingHorizontal": 8,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "width": 40,
                    },
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  櫓
                </Text>
              </View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "justifyContent": "center",
                      "maxWidth": 264,
                      "minWidth": 96,
                    },
                    Object {
                      "maxWidth": 192,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "fontSize": 16,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  Contribute
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
